### PR TITLE
chore: Single Github Action for migration and acceptance tests

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Single Github Action for migration and acceptance tests.

Fixes network and project test groups so acc and mig tests can run together.

Link to any related issue(s): CLOUDP-241162

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/acceptance-tests-runner.yml [hunk 1]
```diff
@@ -103,13 +107,30 @@ env:
   TF_ACC: 1
   TF_LOG: ${{ vars.LOG_LEVEL }}
   ACCTEST_TIMEOUT: ${{ vars.ACCTEST_TIMEOUT }}
-  ACCTEST_REGEX_RUN: "^TestAcc"
+  # Only Migration tests are run when testing a specific previous provider version instead of the latest one
+  ACCTEST_REGEX_RUN: ${{ inputs.provider_version == '' && '^Test(Acc|Mig)' || '^TestMig' }}
```
Comment (lantoli) [inline .github/workflows/acceptance-tests-runner.yml:111]: @EspenAlbert thanks for your suggestion!

File: internal/service/networkcontainer/resource_network_container_migration_test.go [hunk 2]
```diff
@@ -57,13 +59,14 @@ func TestMigNetworkContainer_basicAzure(t *testing.T) {
 
 func TestMigNetworkContainer_basicGCP(t *testing.T) {
 	var (
-		projectID    = acc.ProjectIDExecution(t) // No mig.ProjectIDGlobal because network container
```
Comment (AgustinBettati) [inline internal/service/networkcontainer/resource_network_container_migration_test.go:60]: [q] Now that migration and acceptance test run in the same package execution, is there value in preserving `mig.ProjectIDGlobal`? My understanding is that `ProjectIDExecution` could be used if the resource package is already using it for acceptance tests.

File: .github/workflows/test-suite.yml [hunk 3]
```diff
@@ -57,37 +57,23 @@ jobs:
...
-      matrix:
-        terraform_version: ${{ fromJSON(needs.versions.outputs.terraform_matrix) }}
-    name: acc-tests-${{ matrix.terraform_version || 'latest' }}
+    name: tests-${{ matrix.terraform_version || 'latest' }}-${{ matrix.provider_version || 'latest' }}
     secrets: inherit
     uses: ./.github/workflows/acceptance-tests.yml
     with:
       terraform_version: ${{ matrix.terraform_version }}
+      provider_version: ${{ matrix.provider_version }}
       atlas_cloud_env: ${{ inputs.atlas_cloud_env }}
```
Comment (AgustinBettati) [inline .github/workflows/test-suite.yml:66]: [q] If the test suite is executed with a list of provider versions (for migration tests), acceptance tests would only be run if the value "" is also included right?

Comment (lantoli) [reply .github/workflows/test-suite.yml:66]: correct

File: internal/service/networkcontainer/resource_network_container_migration_test.go (hunk 2 above)
Comment (lantoli) [reply internal/service/networkcontainer/resource_network_container_migration_test.go:60]: you read my mind :-) i was planning a follow-up PR to delete mig.ProjectIDGlobal

**Review by EspenAlbert — APPROVED** [review]
Nice 👍
🤔 would there be a way of sorting the `jobs.change-detection.steps.filter` in `acceptance-tests-runner.yml`?
Did a little smoke test to see if any paths are lost between the `migration-tests.yml` and `acceptance-tests-runner.yml` and it looked good, did you double check?

**Comment by lantoli** [comment]
@EspenAlbert I double-checked but I agree.  I also wanted to do a follow-up PR doing that and also trying to simplify, so for instance we don't need to define twice things like:

```
          generic:
            - 'internal/service/backupcompliancepolicy/*.go'
            - 'internal/service/auditing/*.go'
            - 'internal/service/x509authenticationdatabaseuser/*.go'

          ACCTEST_PACKAGES: |
            ./internal/service/backupcompliancepolicy
            ./internal/service/auditing
            ./internal/service/x509authenticationdatabaseuser

```

**Pull request merged**